lib/src/revset.rs

```diff
@@ -1266,6 +1268,20 @@ static BUILTIN_FUNCTION_MAP: Lazy<HashMap<&'static str, RevsetFunction>> = Lazy:
         };
         Ok(candidates.latest(count))
     });
+    map.insert("immutable", |name, arguments_pair, mut state| {
+        expect_no_arguments(name, arguments_pair)?;
+        if let Some(immutable_heads) = state.immutable_heads {
+            // Avoid infinite recursion if `immutable()` appears in the expression for
+            // immutable heads
+            state.immutable_heads = None;
+            let immutable_heads_expression = parse_program(immutable_heads, state)?;
```

Comment:
need to reset `state.locals`. Otherwise an alias argument would be expanded in the `immutable-heads` definition.

Reply:
I think this is obsolete now that I've made `state.immutable_heads` a `Rc<RevsetExpression>`.